src/main/java/hudson/remoting/Launcher.java

```diff
@@ -624,6 +643,7 @@ private void runAsInboundAgent() throws CmdLineException, IOException, Interrupt
      */
     @SuppressFBWarnings(value = {"CIPHER_INTEGRITY", "STATIC_IV"}, justification = "Integrity not needed here. IV used for decryption only, loaded from encryptor.")
     public List<String> parseJnlpArguments() throws ParserConfigurationException, SAXException, IOException, InterruptedException {
+        initialize();  // For Swarm, or anyone else who invokes this public method directly rather than going through main() or run()
```

Comment:
With lazy initialization of the `noCertificateCheck` functionality, this was working, but only by accident. Now, we explicitly do all initialization in one place, and from every entrypoint.